src/webgpu/api/validation/createBindGroupLayout.spec.ts

```diff
@@ -88,6 +94,96 @@ g.test('mininum_buffer_binding_size')
     }, !success);
   });
 
+g.test('view_dimension')
+  .params(
+    params()
+      .combine(poptions('type', kBindingTypes))
+      .combine(poptions('viewDimension', [undefined, '2d', 'cube'] as const))
+  )
+  .fn(async t => {
+    const { type, viewDimension } = t.params;
+
+    let success = false;
+    if (
+      viewDimension === undefined ||
+      (viewDimension in kTextureViewDimensionInfo && type in kTextureBindingTypeInfo)
+    ) {
+      success = true;
+    }
+
+    t.expectValidationError(() => {
+      t.device.createBindGroupLayout({
+        entries: [{ binding: 0, visibility: GPUShaderStage.COMPUTE, type, viewDimension }],
+      });
+    }, !success);
+  });
+
+g.test('texture_component_type')
+  .params(
+    params()
+      .combine(poptions('type', kBindingTypes))
+      .combine(poptions('textureComponentType', [undefined, 'float', 'uint'] as const))
+  )
+  .fn(async t => {
+    const { type, textureComponentType } = t.params;
+
+    let success = false;
+    if (
+      textureComponentType === undefined ||
+      (textureComponentType in kTextureComponentTypeInfo && type === 'sampled-texture')
+    ) {
+      success = true;
+    }
+
+    t.expectValidationError(() => {
+      t.device.createBindGroupLayout({
+        entries: [{ binding: 0, visibility: GPUShaderStage.COMPUTE, type, textureComponentType }],
+      });
+    }, !success);
+  });
+
+g.test('multisampled')
+  .params(params().combine(poptions('type', kBindingTypes)).combine(pbool('multisampled')))
```

Comment:
3D, Cube, Cube Array, and 2D array textures also cannot be multisampled. We should pass those as viewDimension too. Not sure about 1D textures, but probably not.

Reply:
Probably should have `canMultisample` in `kTextureViewDimensionInfo`, and yes, pretty sure 1D textures can't be multisampled.

Reply:
Good catch. I think we can't support 1D multisample texture, either. But I am not sure whether we can support 2D array multisample texture.  BTW, we haven't described this validation rule in spec. I will submit a PR to add this. And I will submit another test to cover this (or maybe just simply revise this code snippet) once that PR is approved. WDYT? 

This current test is used to cover this statement in spec: `If bindingDescriptor.type is not "sampled-texture", ensure bindingDescriptor.textureComponentType is undefined and bindingDescriptor.multisampled is false.`

Reply:
Multisampled 2D array textures are not supported because on Metal it requires the version of macOS be greater than 10.14. I feel like we may be discussed this somewhere but it's just not in the spec yet. Not sure on whether it's okay to add tests for things that are not yet completely in the spec @kainino0x ?

Reply:
Definitely fine to add things to tests before spec. The tests are easy to change if they're wrong.

Reply:
I will do this with a follow-up CL.